packages/components/src/popover/index.js

```diff
@@ -451,53 +380,33 @@ const Popover = ( {
 		onClickOutside( clickEvent );
 	}
 
-	// Compute the animation position
-	const yAxisMapping = {
-		top: 'bottom',
-		bottom: 'top',
-	};
-	const xAxisMapping = {
-		left: 'right',
-		right: 'left',
-	};
-	const animateYAxis = yAxisMapping[ popoverPosition.yAxis ] || 'middle';
-	const animateXAxis = xAxisMapping[ popoverPosition.xAxis ] || 'center';
-
-	const classes = classnames(
-		'components-popover',
-		className,
-		'is-' + popoverPosition.yAxis,
-		'is-' + popoverPosition.xAxis,
-		{
-			'is-mobile': popoverPosition.isMobile,
-			'is-without-arrow': noArrow || (
-				popoverPosition.xAxis === 'center' &&
-				popoverPosition.yAxis === 'middle'
-			),
-		}
-	);
-
 	// Disable reason: We care to capture the _bubbled_ events from inputs
 	// within popover as inferring close intent.
 
 	let content = (
 		<PopoverDetectOutside onFocusOutside={ handleOnFocusOutside }>
 			<Animate
-				type={ animate && isReadyToAnimate ? 'appear' : null }
-				options={ { origin: animateYAxis + ' ' + animateXAxis } }
+				type={ animate && animateOrigin ? 'appear' : null }
+				options={ { origin: animateOrigin } }
 			>
 				{ ( { className: animateClassName } ) => (
-					<IsolatedEventContainer
```

Comment:
Do you know why we had IsolatedEventContainer and why it's removed now?

Reply:
`IsolatedEventContainer` only stops propagation for `mousedown` events, which I added to the `div` element. I'm not sure why propagation needs to be stopped for that event. There was no comment about it. `IsolatedEventContainer` seems a bit like a strange name for this specific behaviour, which should ideally be commented on in this `Popover` component itself.

Reply:
I think it's probably @aduth's addition. Pinging him just in case.

Reply:
Ok, it looks like the only uses of `IsolatedEventContainer` are `Popover` and `Modal`, and it mentions those, so maybe I should restore it. Too back it creates an extra `div` though.

Reply:
I restored `IsolatedEventContainer`, rewrote it with hooks, and modified it to forward the DOM ref.